readme.md

```diff
@@ -303,6 +303,23 @@ typically yield `SerializableError`s with `exceptionClass=errorCode=<exception c
 user directly. Services should prefer to throw `ServiceException`s instead of the above, since they are easier to
 consume for clients and support transmitting exception parameters in a safe way.
 
+##### `RemoteException` vs `ServiceException` vs `SerializableError` vs `ErrorType`
+
+ - `ErrorType` is a record type, meant to be used as 'compile time constants' - essentially used by services to define the 'enum' of their service exceptions
+ - `SerializableError` is a record type, containing things like error code, error instance id, params - used as the wire format
+ - `ServiceException` is a final subclass of `Exception`, thrown by the server
+ - `RemoteException` is what the client sees if a remote call results in the server internally throwing a `ServiceException`
+
+The rough workflow thus becomes
+
+ - Server code throws an instance of `ServiceException`, containing some `ErrorType`
+ - The `com.palantir.remoting3.servers.jersey.ServiceExceptionMapper` exception mapper
+   - determines the response code for this service exception
+   - converts this into a `SerializableError`
+   - serializes this into the response body as JSON
+ - The client sees a `RemoteException`, which contains the `SerializableException` which was sent over the wire
+ - The client can look at the `SerializableException` and determine what went wrong
+ - Alternatively if the `RemoteException` hits the exception mapper layer, the contained `SerializableError` is sent in the response as-is.
```

Comment:
this needs more explanation to make sense

Reply:
had a go at rephrasing, not sure if any better :/ words are hard